src/m365/entra/commands/pim/pim-role-assignment-remove.ts

```diff
@@ -0,0 +1,250 @@
+      },
+      {
+        option: "--userId [userId]"
+      },
+      {
+        option: "--userName [userName]"
+      },
+      {
+        option: "--groupId [groupId]"
+      },
+      {
+        option: "--groupName [groupName]"
+      },
+      {
+        option: "--administrativeUnitId [administrativeUnitId]"
+      },
+      {
+        option: "--applicationId [applicationId]"
+      },
+      {
+        option: "-j, --justification [justification]"
+      },
+      {
+        option: "--ticketNumber [ticketNumber]"
+      },
+      {
+        option: "--ticketSystem [ticketSystem]"
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.roleDefinitionId && !validation.isValidGuid(args.options.roleDefinitionId)) {
+          return `${args.options.roleDefinitionId} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.applicationId && !validation.isValidGuid(args.options.applicationId)) {
+          return `${args.options.applicationId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['roleDefinitionName', 'roleDefinitionId'] });
+    this.optionSets.push({
+      options: ['userId', 'userName', 'groupId', 'groupName'],
+      runsWhen: (args) => {
+        return args.options.userId !== undefined || args.options.userName !== undefined || args.options.groupId !== undefined || args.options.groupName !== undefined;
+      }
+    });
+    this.optionSets.push({
+      options: ['administrativeUnitId', 'applicationId'],
+      runsWhen: (args) => {
+        return args.options.administrativeUnitId !== undefined || args.options.applicationId !== undefined;
+      }
+    });
+  }
+
+  #initTypes(): void {
+    this.types.string.push('userId', 'userName', 'groupId', 'groupName', 'administrativeUnitId', 'applicationId', 'roleDefinitionName', 'roleDefinitionId', 'justification', 'ticketNumber', 'ticketSystem');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const { userId, userName, groupId, groupName, ticketNumber, ticketSystem } = args.options;
+    try {
+      const token = auth.connection.accessTokens[auth.defaultResource].accessToken;
+      const isAppOnlyAccessToken = accessToken.isAppOnlyAccessToken(token);
+
+      if (isAppOnlyAccessToken) {
```

Comment:
You are missing a condition here. The exception should only be thrown if one of the fields is not filled.

Reply:
Not sure about the condition, but let me know if the updated condition is wrong.

Reply:
I personally need to think for too long when reading this. We should probably do better as follows:

```ts
if (isAppOnlyAccessToken && !userId && !userName && ...etc)
```